pyhub/mcptools/google/sheets/client_async.py

```
@@ -0,0 +1,406 @@
+    def __init__(self):
+        self.agcm: Optional[gspread_asyncio.AsyncioGspreadClientManager] = None
+        self.max_retries = getattr(settings, "GOOGLE_SHEETS_MAX_RETRIES", 5)
+        self._initialized = False
+
+    def _get_credentials_func(self) -> Credentials:
+        """gspread-asyncio용 credentials 함수"""
+        try:
+            return get_sync_credentials()
+        except Exception as e:
+            logger.error(f"Failed to get credentials: {e}")
+            raise AuthenticationError(f"인증 정보를 가져올 수 없습니다: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)